### PR TITLE
Changed log level when PyArrowIterator failed to be imported

### LINKED ISSUE #1271: SNOW-671608: Users are not notified when CAN_USE_ARROW_RESULT_FORMAT is changed
## What is the current behavior?
When defining `CAN_USE_ARROW_RESULT_FORMAT` if we fail to import `PyArrowIterator` for some reason, we are noticed on the **DEBUG** level

## What is the desired behavior?
When defining `CAN_USE_ARROW_RESULT_FORMAT` if we fail to import `PyArrowIterator` for some reason, we want to be noticed on the **ERROR** level

## How would this improve `snowflake-connector-python`?
More visibility, less side-effect

### PR DESCRIPTION
**Pull request by YoniMelki**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #1271

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify the correctness of my new code
   - [X] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Changes in the log level

### CODE REVIEW

**Comment by YoniMelki** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by YoniMelki** [comment]
recheck

**Comment by YoniMelki** [comment]
recheck

**Comment by sfc-gh-sfan** [comment]
I think `warn` is probably better than `error` because query still works though the result format is not arrow. WDYT @sfc-gh-mkeller ?

**Comment by YoniMelki** [comment]
@sfc-gh-sfan Any news here :)?

File: src/snowflake/connector/cursor.py [hunk 1]
```diff
@@ -84,7 +84,7 @@
 
     CAN_USE_ARROW_RESULT_FORMAT = True
 except ImportError as e:  # pragma: no cover
-    logger.debug(
+    logger.error(
```
Comment (sfc-gh-mkeller) [inline src/snowflake/connector/cursor.py:87]:
```suggestion
    logger.warning(
```

**Review by sfc-gh-sfan — APPROVED** [review]

**Comment by YoniMelki** [comment]
@sfc-gh-sfan Can you provide some help for merging? I don't have access to the latest pr-head in jenkins [Details](https://github.com/snowflakedb/snowflake-connector-python/actions/runs/3485059487/jobs/5830245656)

**Comment by sfc-gh-sfan** [comment]
Merged! Thanks for reminding me.

**Pull request merged**